projects/packages/videopress/src/client/lib/video-tracks/index.ts

```diff
@@ -19,6 +24,56 @@ export const TRACK_KIND_OPTIONS = [
 const { siteType = '' } = window?.videoPressEditorState || {};
 const shouldUseJetpackVideoFetch = siteType !== 'simple';
 
+type isAutogeneratedChaterFileParamsProps = {
+	guid?: VideoGUID;
+	isPrivate?: boolean;
+};
+
+/**
+ * Load the file content, and check if it's autogenerated,
+ * based on the `videopress-chapters-auto-generated` comment.
+ * The function will try to anticipate the video privacy,
+ * based on the block attributes.
+ * If the first request fails, it will try again with the token.
+ *
+ * @param {string} fileUrl                              - the track file url
+ * @param {isAutogeneratedChaterFileParamsProps} params - function parameters
+ * @returns {Promise<boolean>}   true if the file is autogenerated.
+ */
+export async function isAutogeneratedChapterFile(
+	fileUrl: string,
+	params?: isAutogeneratedChaterFileParamsProps
+): Promise< boolean > {
+	if ( ! fileUrl ) {
+		return false;
+	}
+
+	let tokenData: null | MediaTokenProps;
+	let queryString = '';
+
+	// Try to anticipate the video privacy, based on the block attributes.
+	if ( params.isPrivate ) {
+		tokenData = await getMediaToken( 'playback', { guid: params.guid } );
+		queryString = '?' + new URLSearchParams( { metadata_token: tokenData?.token } ).toString();
+	}
+
+	let response = await fetch( fileUrl + queryString );
+
+	// If the file is private, and response is 403, try with the token.
+	if ( ! response.ok && response.status === 403 && params.guid ) {
```

Comment:
Good point. Even if the `param.isPrivate` is `false`, by receiving a `403` we have an indication that the privacy is set to private on the backend.